src/sinks/datadog/logs/service.rs

```diff
@@ -90,16 +92,38 @@ impl DriverResponse for LogApiResponse {
 pub struct LogApiService {
     client: HttpClient,
     uri: Uri,
-    enterprise: bool,
+    user_provided_headers: IndexMap<HeaderName, HeaderValue>,
 }
 
 impl LogApiService {
-    pub const fn new(client: HttpClient, uri: Uri, enterprise: bool) -> Self {
-        Self {
+    pub fn new(
+        client: HttpClient,
+        uri: Uri,
+        headers: IndexMap<String, String>,
+    ) -> crate::Result<Self> {
+        let headers = Self::validate_headers(&headers)?;
+
+        Ok(Self {
             client,
             uri,
-            enterprise,
+            user_provided_headers: headers,
+        })
+    }
+
+    fn validate_headers(
```

Comment:
Nit: this looks pretty similar to the implementation in `src/sinks/http.rs`. I wonder if we could extract it?

Reply:
👍 Went ahead and moved the common validation bits to `util/http.rs`: https://github.com/vectordotdev/vector/pull/15745/commits/630cebc64f4e36fbbc6515e055fd1024623716fc. Let me know how that looks.

Reply:
Looks good, thanks for refactoring that!